in_toto/user_settings.py

```diff
@@ -98,35 +115,93 @@ def get_env():
 
 
 def get_rc():
-  """Load RCfiles from the usual places and return a dictionary of
-  key value pairs.
+  """
+  <Purpose>
+    Reads RCfiles from the paths defined in `RC_PATHS` and returns
+    a dictionary with all parsed key-value pairs.
+
+    The RCfile format is as expected by Python's builtin `ConfigParser` with
+    the addition that values that contain colons (:) are split at the position
+    of the colons and converted into a list.
+
+    Section titles in RCfiles are ignored when parsing the key-value pairs.
+    However, there has to be at least one section defined.
+
+    The paths in `RC_PATHS` are ordered in reverse precedence, i.e. each file's
+    settings override a previous file's settings, e.g. a setting defined
+    in `.in_totorc` (in the current working dir) overrides the same
+    setting defined in `~/.in_totorc` (in the user's home dir) and so on ...
+
+    Example:
+
+    ```
+    # E.g. file `.in_totorc` in current working directory
+    [in-toto setting]
+    ARTIFACT_BASE_PATH = /home/user/project
+    ARTIFACT_EXCLUDES = *.link:.gitignore
+    ```
+
+    produces
+
+    ```
+    {
+      "ARTIFACT_BASE_PATH": "/home/user/project"
+      "ARTIFACT_EXCLUDES": ["*.link", ".gitignore"]
+    }
+    ```
 
-  - RCfile format as expected by builtin ConfigParser
-  - The format dictates that there is at least one section,
-    however we are only interested in the values and ignore the sections
-  - Values that contain colons are converted into a list
-  - Variable names are converted to upper case
+  <Exceptions>
+    None.
+
+  <Side Effects>
+    Calls function to read files from disk.
```

Comment:
Another instance where someone might argue that this isn't really a side effect.